### PR TITLE
Firefly-1545: charts better handle large selections

### PR DESCRIPTION
**Pull request by robyww**
#### [Firefly-1545](https://jira.ipac.caltech.edu/browse/FIREFLY-1545): charts better handle large selections

 - charts now give warning when selection over 30000 and will not show the selection

#### Testing
 - https://fireflydev.ipac.caltech.edu/firefly-1545-charts/firefly
 - Load a big file (over 30000 rows)
 - it will show a heatmap so make a scatter chart
 - Select a large area and click check icon
 - A warning dialog will appear only the first time
 - The toolbar will show a warning icon when scatter selection is over 30000

### CODE REVIEW

File: src/firefly/js/charts/ui/PlotlyToolbar.jsx [hunk 1]
```diff
@@ -62,22 +62,33 @@ function getToolbarStates(chartId) {
...
+
+
     const hasSelectionMode = Boolean(tbl_id);
     const help_id = getChartData(chartId)?.help_id;
 
     return (
         <Stack direction='row' alignItems='center'>
+            {isScatter && selCount>MAX_CHART_SELECT&&
+                <SelectionWarningBtn
+                    warningStr={`Scatter chart cannot show selection above ${MAX_CHART_SELECT}, currently there are ${selCount} points selected but not shown not the chart`}/>}
```
Comment (gpdf) [inline src/firefly/js/charts/ui/PlotlyToolbar.jsx:89]: This message needs some editing; discussing this on Slack.  I'll follow up with a specific recommendation here later.

File: src/firefly/js/charts/ui/PlotlyToolbar.jsx [hunk 2]
```diff
@@ -302,6 +313,18 @@ function OptionsBtn({chartId}) {
...
 }
 
+function SelectionWarningBtn({warningStr}) {
+    return (
+        <WarningButton onClick={() => showInfoPopup(warningStr, 'Selection Warning')} tip={warningStr}/>
+    );
+}
+
+const showMaxSelectionWarningOnce = once(() => {
+    const str= `The scatter chart cannot show selection above ${MAX_CHART_SELECT}, the data is still selected just not shown in the chart`;
```
Comment (gpdf) [inline src/firefly/js/charts/ui/PlotlyToolbar.jsx:323]: This message will also need editing.

**Review by gpdf — CHANGES_REQUESTED** [review]
The warning messages need some polishing.  I haven't had a chance to actually run the branch build yet, but I wanted to at least get this in.

**Comment by loitly** [comment]
I don't like how this is implemented.  I offer an alternative solution that addresses the root cause, though I believe further optimization is possible if needed. I strongly recommend reviewing and considering this alternative approach:
https://github.com/Caltech-IPAC/firefly/pull/1631

**Pull request closed without merging**